src/snowflake/snowpark/dataframe.py

```diff
@@ -666,13 +705,48 @@ def collect_nowait(
         See also:
             :meth:`collect()`
         """
+        if _emit_ast:
+            # Add an Assign node for SpDataframeCollect()
+            repr = self._session._ast_batch.assign()
+            expr = with_src_position(repr.expr.sp_dataframe_collect_no_wait)
```

Comment:
I suspect we should just add `no_wait: Bool` to `SpDataframeCollect` and factor out the code here.